operator/internal/manifests/mutate.go

```diff
@@ -47,6 +47,16 @@ func MutateFuncFor(existing, desired client.Object) controllerutil.MutateFn {
 			wantCm := desired.(*corev1.ConfigMap)
 			mutateConfigMap(cm, wantCm)
 
+		case *corev1.Secret:
+			s := existing.(*corev1.Secret)
+			wantS := desired.(*corev1.Secret)
+			mutateSecret(s, wantS)
+			existingAnnotations := s.GetAnnotations()
+			if err := mergeWithOverride(&existingAnnotations, depAnnotations); err != nil {
```

Comment:
For streamlining error reporting reasons we decided to use `mergo.Merge` for the mutation functions to keep some sort of uniformity instead of writing custom for loops. For the sake of homogeneity i keep using the pattern here.